app.js

```diff
@@ -9,65 +9,101 @@ const reportEl = document.getElementById('report');
-
+    headEl.style.backgroundImage = `url("./assets/${head}-head.png")`;
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
 
 middleDropdown.addEventListener('change', () => {
     // get the value of the middle dropdown
-
+    const middle = middleDropdown.value;
     // increment the middle change count state
-    
+    middleCount++;
     // update the dom for the middle (NOTE: use style.backgroundImage on the middleEl div instead of trying to set the .src -- it's NOT an img tag!)
-
+    middleEl.style.backgroundImage = `url("./assets/${middle}-middle.png")`;
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
 
 bottomDropdown.addEventListener('change', () => {
     // get the value of the bottom dropdown
-
+    const bottom = bottomDropdown.value;
     // increment the bottom change count state
-    
+    bottomCount++;
     // update the dom for the bottom (NOTE use style.backgroundImage on the bottomEl div instead of trying to set the .src -- it's NOT an img tag!)
-
+    bottomEl.style.backgroundImage = `url("./assets/${bottom}-pants.png")`;
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
 catchphraseButton.addEventListener('click', () => {
     // get the value of the catchphrase input
-    
+    const newCatchPhrase = catchphraseInput.value;
     // push the new catchphrase to the catchphrase array in state
-
+    catchPhrases.push(newCatchPhrase);
     // clear out the form input's value so it's empty to the user
-   
+    catchphraseInput.value = '';
     // update the dom to show the new catchphrases (refactor to/call displayCatchphrases to do this work)
-
+    displayCatchphrases();
 });
 
 function displayStats() {
     // text content of the reportEl to tell the user how many times they've changed each piece of the state
+    reportEl.textContent = `You have changed the head ${headCount} times, the middle ${middleCount} times and the bottom ${bottomCount} times!`; 
 }
 
 function displayCatchphrases() {
     // clear out the DOM for the currently displayed catchphrases
-
+    catchphrasesEl.textContent = '';
     // loop through each catchphrase in state
-   
+    for (let catchphrase of catchPhrases) {
+        const div = document.createElement('div');
+        
+        div.classList.add('catchphrase');
     // and for each catchphrase
     
     // create an HTML element with the catchphrase as its text content
-    
+        div.textContent = catchphrase;
     // and append that HTML element to the cleared-out DOM
+        catchphrasesEl.append(div);
+    }
 }
+
+nameButton.addEventListener('click', ()=> {
+    const newName = nameInput.value;
+    names.push(newName);
+    nameInput.value = '';
+    displayName();
+});
+
+function displayName() {
```

Comment:
totally a stretch goal so my feedback is irrelevant but do you want this to be displaying all the names like the catchphrases or do you want it to be just one name at a time?